src/api/hooks/useApiQuery.ts

```diff
@@ -21,7 +21,10 @@ function useApiQuery<TData>(
       }
       return queryFn(api);
     },
-    options,
+    {
+      ...options,
+      enabled: options?.enabled ? options.enabled && api?.isConnected : api?.isConnected,
```

Comment:
This will work fine if we pass `enabled: true` or if we don't pass `enabled` at all, but if we pass `enabled: false` and the api is connected it will execute the query function when it shouldn’t. 

One way to solve this could be: 

1. Add default value: 
      `options: UseQueryOptions<TData> = { enabled: true },`
2. then 
    `enabled: options.enabled && Boolean(api?.isConnected)`

Reply:
done ✅